frontend/integration-tests/tests/kubevirt/non-admin.scenario.ts

```diff
@@ -0,0 +1,140 @@
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
+
+      await click(crudView.createItemButton);
+      await click(createWithYAMLLink);
+      await yamlView.isLoaded();
+    }
+
+    await browser.get(`${appHost}/k8s/ns/${ns}/vmtemplates`);
+    if (perm === 'noAccess') {
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
+    }
+
+    await browser.get(`${appHost}/overview/ns/${ns}/`);
+    if (perm === 'noAccess') {
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+    }
+  };
+
+  const verifyPermissionWithRoleBinding = async function( perm: string, ns: string, roleBinding: string ) {
+    execSync(`oc adm policy add-${roleBinding}-to-user ${perm} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+    await verifyPermissions(perm, ns);
+    execSync(`oc adm policy remove-${roleBinding}-from-user ${perm} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+  };
+
+  beforeAll(async() => {
+    try {
+      execSync('oc get -o yaml -n openshift-config secret test');
+    } catch (error) {
+      createResources([nonAdminSecret, nonAdminProvider]);
+    }
+  });
+
+  afterAll(async() => {
+    await logout();
+    await browser.wait(until.presenceOf($('.login-pf')));
+    await loginView.login(KUBEADMIN_IDP, KUBEADMIN_USERNAME, BRIDGE_KUBEADMIN_PASSWORD);
+  });
+
+  it('Login with nonadmin', async() => {
+    await logout();
+    // wait for logout complete
+    await browser.sleep(10000);
+    // TODO: remove the refill appHost after bug 1721423 is fixed.
+    await browser.get(appHost);
+    await browser.wait(until.presenceOf($('.login-pf')));
+    await loginView.login(BRIDGE_HTPASSWD_IDP, BRIDGE_HTPASSWD_USERNAME, BRIDGE_HTPASSWD_PASSWORD);
+    //expect(loginView.userDropdown.getText()).toContain(BRIDGE_HTPASSWD_IDP);
+  });
+
+  it(`Creates test namespace ${nonAdminNS} for nonAdmin tests`, async() => {
```

Comment:
shouldn't this be rather in beforeAll setup phase? Or why is the `exists` check necessary?

Reply:
It can't in beforeAll as non-admin isn't login yet.  The block is basically a copy of `base.scenario.ts`

Reply:
Ok so we do not need to check if the namespace exists. Shouldn't we also cleanup the namespace?